File: 1-easy/problems/3000-maximum-area-of-longest-diagonal-rectangle/solution.ts
function areaOfMaxDiagonal(dimensions: any): boolean | number | string | any {
  """
  :type dimensions: List[List[int]]
  :rtype: int
  """
  return max((l**2+w**2, l*w) for l, w in dimensions)[1];
}

export { areaOfMaxDiagonal };

File: 1-easy/problems/3000-maximum-area-of-longest-diagonal-rectangle/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def areaOfMaxDiagonal(self, dimensions):
        """
        :type dimensions: List[List[int]]
        :rtype: int
        """
        return max((l**2+w**2, l*w) for l, w in dimensions)[1]
